src/Features/Core/Portable/Shared/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -6,57 +6,24 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static class DiagnosticAnalyzerExtensions
     {
-        public static DiagnosticAnalyzerCategory GetDiagnosticAnalyzerCategory(this DiagnosticAnalyzer analyzer)
-        {
-            var category = DiagnosticAnalyzerCategory.None;
-
-            if (analyzer is DocumentDiagnosticAnalyzer)
-            {
-                category |= DiagnosticAnalyzerCategory.SyntaxAnalysis | DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
-            }
-            else if (analyzer is ProjectDiagnosticAnalyzer)
-            {
-                category |= DiagnosticAnalyzerCategory.ProjectAnalysis;
-            }
-            else
-            {
-                if (analyzer is IBuiltInAnalyzer builtInAnalyzer)
-                {
-                    category = builtInAnalyzer.GetAnalyzerCategory();
-                }
-                else
-                {
-                    // It is not possible to know the categorization for a public analyzer,
-                    // so return a worst-case categorization.
-                    category = (DiagnosticAnalyzerCategory.SyntaxAnalysis | DiagnosticAnalyzerCategory.SemanticDocumentAnalysis | DiagnosticAnalyzerCategory.ProjectAnalysis);
-                }
-            }
-
-            return category;
-        }
-
         public static bool SupportsSyntaxDiagnosticAnalysis(this DiagnosticAnalyzer analyzer)
         {
-            var category = analyzer.GetDiagnosticAnalyzerCategory();
-            return (category & DiagnosticAnalyzerCategory.SyntaxAnalysis) != 0;
+            return !(analyzer is ProjectDiagnosticAnalyzer);
         }
 
         public static bool SupportsSemanticDiagnosticAnalysis(this DiagnosticAnalyzer analyzer)
         {
-            var category = analyzer.GetDiagnosticAnalyzerCategory();
-            return (category & (DiagnosticAnalyzerCategory.SemanticSpanAnalysis | DiagnosticAnalyzerCategory.SemanticDocumentAnalysis)) != 0;
+            return !(analyzer is ProjectDiagnosticAnalyzer);
```

Comment:
this is not a complain about your PR.  But if this goes through i think i'd like to simplify things further as we can merge SupportsSyntaxDiagnosticAnalysis and SupportsSemanticDiagnosticAnalysis.  It would interesting to see how that simplification would bubble outwards.

Reply:
Didn't actually go that far unfortunately.  You can see my attempt over here: https://github.com/CyrusNajmabadi/roslyn/tree/remove-analyzer-category

Feel free ot merge in if you feel it adds value.